### ISSUE TITLE
Seg fault on closing napari with framework launch mac

### ISSUE THREAD

**Issue by sofroniewn** [napari/napari#1618]
## 🐛 Bug

We're currently getting a seg fault when quitting napari after it has been launched with the framework build on mac - likely after #1554. See

```
(base) czirwc1macos2701:napari nsofroniew$ napari
/Users/nsofroniew/opt/anaconda3/bin/pythonw: line 3: 34762 Segmentation fault: 11  /Users/nsofroniew/opt/anaconda3/python.app/Contents/MacOS/python "$@"
```

If you do this a couple times in a row you start getting this dialog which blocks launch until it is addressed.

<img width="532" alt="Screen Shot 2020-09-04 at 11 14 21 AM" src="https://user-images.githubusercontent.com/6531703/92273259-e2c51900-ee9f-11ea-8267-443f7fc495a0.png">

Fixing this and the minimum version issues seems like a critical thing we should do before the 0.3.8 release
cc @jni @tlambert03 @Czaki

**Comment by Czaki** [comment]
Could You provide more details? MacOS version, python version? 
I cannot reproduce this on my Mac.

**Comment by sofroniewn** [comment]
macOS Catalina, Version `10.15.5` and Python 3.7.4

napari --info is

```
napari: 0.3.7.dev65+g761a22ce
Platform: Darwin-19.5.0-x86_64-i386-64bit
Python: 3.7.0 (default, Jun 28 2018, 07:39:16) [Clang 4.0.1 (tags/RELEASE_401/final)]
Qt: 5.14.1
PySide2: 5.14.1
NumPy: 1.19.1
SciPy: 1.4.1
Dask: 2.23.0
VisPy: 0.6.5.dev2+g9e0822db

GL version: 2.1 INTEL-14.6.18
MAX_TEXTURE_SIZE: 16384
```

**Comment by Czaki** [comment]
I install anaconda and still cannot reproduce this error. My system is 10.14.6. 


```
➜  ~ napari --info                       
napari: 0.3.8.dev1+gce6f2267
Platform: macOS-10.14.6-x86_64-i386-64bit
Python: 3.8.3 (default, Jul  2 2020, 11:26:31)  [Clang 10.0.0 ]
Qt: 5.14.1
PyQt5: 5.14.1
NumPy: 1.18.5
SciPy: 1.5.0
Dask: 2.20.0
VisPy: 0.6.4
```

**Comment by sofroniewn** [comment]
Maybe it is Pyside2 vs PyQt5? I can try that too. Thanks for trying @Czaki!!!

**Comment by sofroniewn** [comment]
Yes! I only get error with PySide2 5.14.1 and NOT PyQt5 5.14.1

**Comment by Czaki** [comment]
Sometimes ago I have symilar problem and it could be PySide. In my case it was problem with wrong cleaning order. 

How install PySide with Anaconda (I use mainly core CPython )

**Comment by sofroniewn** [comment]
Do you think @Czaki it is how I installed PySide2 or is there for all PySide2 users?

Do you think there is a fix we can make to prevent this crash?

**Comment by Czaki** [comment]
I cannot simply reproduce because #1262 broke #1080

**Comment by Czaki** [comment]
Also work:

```
napari: 0.3.8.dev1+gce6f2267
Platform: macOS-10.14.6-x86_64-i386-64bit
Python: 3.8.3 (default, Jul 2 2020, 11:26:31) [Clang 10.0.0 ]
Qt: 5.14.1
PySide2: 5.14.1
NumPy: 1.18.5
SciPy: 1.5.0
Dask: 2.20.0
VisPy: 0.6.4

GL version: 2.1 ATI-2.11.21
MAX_TEXTURE_SIZE: 16384

Plugins:
- napari_plugin_engine: 0.1.7
- svg: 0.1.4
```

**Comment by sofroniewn** [comment]
> I cannot simply reproduce because #1262 broke #1080

Ah, I guess this is issue #1349 that you are referring too. @Czaki can you maybe post on #1349 about how this has broken #1080 and has made switching PyQt5/ PySide2 harder for you. Maybe @pwinston can spend some time looking into fixing that one.

**Comment by sofroniewn** [comment]
Interesting that you could get PySide2 working @Czaki!

Maybe the Python 3.8 vs 3.7 matters too. We'll probably need to get some more eyeballs on this one to really understand, cc @ziyangczi

**Comment by Czaki** [comment]
If I good remember when I have symilar problem I run whole program under c debuger 
```
lldb  -- `pyenv which python` -m pytest .
```
and trace source of problem 

@sofroniewn 
It is different problem. #1080 introduce check that if `pyrcc5` fails then `pyside2-rcc` is called. 
When working with `pyenv` ones `pyrcc5` is installed such named file in in path. But if current environment does not contain it then produce such message:

```
pyenv: pyrcc5: command not found

The `pyrcc5' command exists in these Python versions:
  3.6.9/envs/partseg3.6
  3.7.4/envs/napari-pyqt5
  3.7.7/envs/partseg3.7
  napari-pyqt5
  partseg3.6
  partseg3.7
```

**Comment by sofroniewn** [comment]
> If I good remember when I have symilar problem I run whole program under c debuger

Ok thanks for that tip, I can try that right now

> It is different problem. #1080 introduce check that if pyrcc5 fails then pyside2-rcc is called.

Is this something that you can fix @Czaki? If so we'd happily merge a PR. Apologies we broke #1080, if you could also add a test that would have caught the bug introduced in #1262 then maybe we'll be able to avoid doing it again!!

**Comment by Czaki** [comment]
I wil create such PR. This time with tests.

**Comment by sofroniewn** [comment]
Ok, testing this in a clean Python 3.7 environment with napari 0.3.7, PySide2, and pythonw works fine. I'm sorry for not doing that sooner! I guess something is wrong with my other environment, which has loads of stuff in it!! I can try looking into that more. Thanks @Czaki for testing!!

**Comment by Czaki** [comment]
I see that you have developer vispy version in main env. Maybe this is source?

**Comment by sofroniewn** [comment]
This ended up not being reproducible and has resolved when I created a new environment. I am closing now, but if people encounter this more frequently we can reopen or discuss on a new issue
